### PR TITLE
install function core tools

### PR DESCRIPTION
**Pull request by ivywei0125**
What does this implement/fix? Explain your changes.
---------------------------------------------------
function-core-tools-warning
function-core-tools-error
rider-azure-functions

Does this close any currently open issues?
------------------------------------------

Any relevant logs, screenshots, error output, etc.?
-------------------------------------

Any other comments?
-------------------
…

Has this been tested?
---------------------------
- [ ] Tested

### CODE REVIEW

File: PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/ui/AzurePanel.form [hunk 1]
```diff
@@ -187,39 +187,24 @@
...
       </component>
-      <component id="28687" class="javax.swing.JLabel" binding="funcCoreToolsDownloadPathLabel">
+      <component id="7db42" class="com.intellij.ui.components.ActionLink" binding="installFuncCoreToolsAction" custom-create="true">
         <constraints>
-          <grid row="7" column="0" row-span="1" col-span="1" vsize-policy="0" hsize-policy="0" anchor="8" fill="0" indent="0" use-parent-layout="false"/>
+          <grid row="7" column="1" row-span="1" col-span="1" vsize-policy="0" hsize-policy="0" anchor="4" fill="0" indent="0" use-parent-layout="false"/>
         </constraints>
         <properties>
-          <text value="Default download path:"/>
+          <text value="Install latest version"/>
```
Comment (ivywei0125) [inline PluginsAndFeatures/azure-toolkit-for-intellij/src/main/java/com/microsoft/intellij/ui/AzurePanel.form]: Install the latest version

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java [hunk 2]
```diff
@@ -69,16 +68,16 @@ private void cacheReleaseInfoFromFeed(ReleaseFilterProvider filterProvider) {
...
 
-    private void doDownloadRelease(ReleaseInfo releaseInfo, String downloadDirPath) {
-        final File tempFile;
+    private void doDownloadRelease(@Nullable ReleaseInfo releaseInfo, String downloadDirPath) {
+        if (Objects.isNull(releaseInfo)) {
+            return;
+        }
         try {
-            tempFile = File.createTempFile(
+            final File tempFile = File.createTempFile(
```
Comment (ivywei0125) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java:84]: check if delete automatically

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java [hunk 3]
```diff
@@ -201,6 +207,34 @@ public void registerHandlers(AzureActionManager am) {
...
+                    @Override
+                    public void onSuccess() {
+                        final Action<Object> openSettingsAction = AzureActionManager.getInstance().getAction(ResourceCommonActionsContributor.OPEN_AZURE_SETTINGS);
+                        final Action<Object> openSettingsActionInMessage = new Action<>(Action.Id.of("common.open_azure_settings_dialog"), new ActionView.Builder("Open Azure Settings")) {
+                            @Override
+                            public void handle(Object source, Object e) {
+                                AzureTaskManager.getInstance().runLater(() -> openSettingsAction.handle(null, e));
+                            }
+                        };
+                        final String INSTALL_SUCCEED_MESSAGE = "download and install functions core tools successfully.";
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java]: Azure Functions Core Tools.

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java [hunk 4]
```diff
@@ -120,6 +123,28 @@ private void createIfNotExist(String dirPath) {
         dstDir.mkdirs();
     }
 
+    /**
+     * refer to https://github.com/JetBrains/azure-tools-for-intellij
+     * */
+    private ReleaseFilter generateFilter() {
```
Comment (ivywei0125) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java:131]: ignore sensitive case

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java [hunk 5]
```diff
@@ -201,6 +207,34 @@ public void registerHandlers(AzureActionManager am) {
...
+                        AzureMessager.getMessager().success(INSTALL_SUCCEED_MESSAGE, "Install succeed", openSettingsActionInMessage);
+                    }
+
+                    @Override
+                    public void onFail() {}
+                };
+                AzureTaskManager.getInstance().runInModal("Download and Install Functions Core Tools",
```
Comment (wangmingliang-ms) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java]: Azure Functions Core Tools.

File: PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java [hunk 6]
```diff
@@ -201,6 +207,34 @@ public void registerHandlers(AzureActionManager am) {
...
+                        () -> FunctionsCoreToolsManager.getInstance().downloadReleaseWithFilter(installPath, listener));
+            }));
+        };
+        am.registerHandler(FunctionAppActionsContributor.DOWNLOAD_CORE_TOOLS, downloadFuncCoreToolsHandler);
```
Comment (ivywei0125) [inline PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-appservice-java/src/main/java/com/microsoft/azure/toolkit/intellij/appservice/AppServiceIntelliJActionsContributor.java:220]: new exception

File: Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java [hunk 7]
```diff
@@ -86,12 +87,21 @@ private void doDownloadRelease(@Nullable ReleaseInfo releaseInfo, String downloa
             unzip(tempFile, Paths.get(downloadDirPath, releaseInfo.releaseVersion).toString());
         } catch (final Exception e) {
             AzureMessager.getMessager().error(e, INSTALL_FAILED_MESSAGE);
+            sendTelemetryWhenFail();
```
Comment (ivywei0125) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java]: get message from exception

Comment (wangmingliang-ms) [inline Utils/azure-toolkit-ide-libs/azure-toolkit-ide-appservice-lib/src/main/java/com/microsoft/azure/toolkit/ide/appservice/function/coretools/FunctionsCoreToolsManager.java]: throw exception directly and let @AzureOperation handle failure and telemetry

**Review by wangmingliang-ms — APPROVED** [review]
LGTM

**Pull request merged**